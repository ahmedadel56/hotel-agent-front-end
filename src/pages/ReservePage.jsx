/* eslint-disable */
import React, { useState, useEffect } from 'react';
import { useForm, Controller } from 'react-hook-form';
import { useSelector, useDispatch } from 'react-redux';
import ReactDatePicker from 'react-datepicker';
import { addHotelReservation } from '../redux/Reservations/Reservation';
import 'react-datepicker/dist/react-datepicker.css';
import countries from '../components/CountryPicker';
import background from '../img/background.jpg';
import Alert from '../components/Alert';

const ReservePage = () => {
  const dispatch = useDispatch();
  const [success, setSucces] = useState(false);
  useEffect(() => {
    if (success) {
    const timeId = setTimeout(() => {
      setSucces(false)
    }, 1500)

    return () => {
      clearTimeout(timeId)
    }
    }
  }, [success]);

  const {
    register, formState: { errors }, handleSubmit, control,
  } = useForm({
    defaultValues: {
      user_id: localStorage.getItem('userId'),
    },
  });
  const onSubmit = (data, e) => {
    dispatch(addHotelReservation(data));
    e.target.reset();
    setSucces(true);
  };

  const { data } = useSelector((state) => state.MostRecent);

  const countryToFlag = (isoCode) => (typeof String.fromCodePoint !== 'undefined'
    ? isoCode
      .toUpperCase()
      .replace(/./g, (char) => String.fromCodePoint(char.charCodeAt(0) + 127397))
    : isoCode);

  return (
    <><>
      {success ? (<Alert type="success">
        <p>Reservation added successfully</p>
      </Alert>) : null}
    </><div
      className="reserve-container"
      style={{
        backgroundImage: `url(${background})`,
      }}
    >
        <>
          <div className="reserve-heading">
            <h1>BOOK A HOTEL</h1>
            <hr />
            <p>Please select a Hotel</p>
          </div>
          <div className="reserve-form">
            <form onSubmit={handleSubmit(onSubmit)}>
              <section className="countries-container">
                <select {...register('country', { required: true })}>
                  {countries.map((item) => (
                    <option key={item.label} value={item.label}>
                      {item.label}
                      {' '}
                      {countryToFlag(item.code)}
                    </option>
                  ))}
                </select>
              </section>
              <section className="date-container">
                <Controller
                  control={control}
                  name="date"
                  rules={{ required: true }}
                  render={({
                    field: {
                      onChange, onBlur, value, ref,
                    },
                  }) => (
                    <ReactDatePicker
                      className="date-picker"
                      onChange={onChange}
                      onBlur={onBlur}
                      selected={value}
                      placeholderText="Select date" />
                  )} />
                {errors.date?.type === 'required' && <span>Date is required</span>}
              </section>
              <section className="options-container">
                <div className="login-form">
                  <input
                    placeholder="🏖 Number of days"
                    {...register('number_of_days', {
                      required: true,
                      min: 1,
                      pattern: /^[0-9]*$/,
                    })} />
                    <p>
                  {errors.number_of_days?.type === 'required' && <span>Number of days is required</span>}
                  {errors.number_of_days?.type === 'min' && (
                    <span>Number of days must be more than 0</span>
                  )}
                  {errors.number_of_days?.type === 'pattern' && (
                    <span>Number of days must be numeric</span>
                  )}
                  </p>
                </div>
                <div className="login-form">
                  <input
                    placeholder="🏘 Number of rooms"
                    {...register('number_of_rooms', {
                      required: true,
                      min: 1,
                      pattern: /^[0-9]*$/,
                    })} />
                    <p>
                  {errors.number_of_rooms?.type === 'required' && <span>Number of rooms is required</span>}
                  {errors.number_of_rooms?.type === 'min' && (
                    <span>Number of rooms must be more than 0</span>
                  )}
                  {errors.number_of_rooms?.type === 'pattern' && (
                    <span>Number of rooms must be numeric</span>
                  )}
                  </p>
                </div>
                <div className="login-form">
                  <input
                    placeholder="👥 Number of guests"
                    {...register('number_of_guests', {
                      required: true,
                      min: 1,
                      pattern: /^[0-9]*$/,
                    })} />
                    <p>
                  {errors.number_of_guests?.type === 'required' && <span>Number of guests is required</span>}
                  {errors.number_of_guests?.type === 'min' && (
                    <span>Number of guests must be more than 0</span>
                  )}
                  {errors.number_of_guests?.type === 'pattern' && (
                    <span>Number of guests must be numeric</span>
                  )}
                  </p>
                </div>
              </section>
              <section className="select-hotel-container">
                <label>Select Hotel:</label>
                <select {...register('hotel_id', { required: true })}>
                  {data.map((item) => (
                    <option key={item.name} value={item.id}>
                      {item.name}
                    </option>
                  ))}
                </select>
              </section>
              <section className="submit-container">
                <input type="submit" value="RESERVE" />
              </section>
            </form>
          </div>
        </>
      </div></>
  );
};

export default ReservePage;
