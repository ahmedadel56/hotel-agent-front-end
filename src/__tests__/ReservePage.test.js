import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import ReservePage from '../pages/ReservePage';

it('Check if Reserve page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <ReservePage />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
