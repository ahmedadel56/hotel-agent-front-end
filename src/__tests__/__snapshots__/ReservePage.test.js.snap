// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Reserve page renders correctly 1`] = `
<div
  className="reserve-container"
  style={
    Object {
      "backgroundImage": "url(background.jpg)",
    }
  }
>
  <div
    className="reserve-heading"
  >
    <h1>
      BOOK A HOTEL
    </h1>
    <hr />
    <p>
      Please select a Hotel
    </p>
  </div>
  <div
    className="reserve-form"
  >
    <form
      onSubmit={[Function]}
    >
      <section
        className="countries-container"
      >
        <select
          name="country"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option
            value="Andorra"
          >
            Andorra
             
            🇦🇩
          </option>
          <option
            value="United Arab Emirates"
          >
            United Arab Emirates
             
            🇦🇪
          </option>
          <option
            value="Afghanistan"
          >
            Afghanistan
             
            🇦🇫
          </option>
          <option
            value="Antigua and Barbuda"
          >
            Antigua and Barbuda
             
            🇦🇬
          </option>
          <option
            value="Anguilla"
          >
            Anguilla
             
            🇦🇮
          </option>
          <option
            value="Albania"
          >
            Albania
             
            🇦🇱
          </option>
          <option
            value="Armenia"
          >
            Armenia
             
            🇦🇲
          </option>
          <option
            value="Angola"
          >
            Angola
             
            🇦🇴
          </option>
          <option
            value="Antarctica"
          >
            Antarctica
             
            🇦🇶
          </option>
          <option
            value="Argentina"
          >
            Argentina
             
            🇦🇷
          </option>
          <option
            value="American Samoa"
          >
            American Samoa
             
            🇦🇸
          </option>
          <option
            value="Austria"
          >
            Austria
             
            🇦🇹
          </option>
          <option
            value="Australia"
          >
            Australia
             
            🇦🇺
          </option>
          <option
            value="Aruba"
          >
            Aruba
             
            🇦🇼
          </option>
          <option
            value="Alland Islands"
          >
            Alland Islands
             
            🇦🇽
          </option>
          <option
            value="Azerbaijan"
          >
            Azerbaijan
             
            🇦🇿
          </option>
          <option
            value="Bosnia and Herzegovina"
          >
            Bosnia and Herzegovina
             
            🇧🇦
          </option>
          <option
            value="Barbados"
          >
            Barbados
             
            🇧🇧
          </option>
          <option
            value="Bangladesh"
          >
            Bangladesh
             
            🇧🇩
          </option>
          <option
            value="Belgium"
          >
            Belgium
             
            🇧🇪
          </option>
          <option
            value="Burkina Faso"
          >
            Burkina Faso
             
            🇧🇫
          </option>
          <option
            value="Bulgaria"
          >
            Bulgaria
             
            🇧🇬
          </option>
          <option
            value="Bahrain"
          >
            Bahrain
             
            🇧🇭
          </option>
          <option
            value="Burundi"
          >
            Burundi
             
            🇧🇮
          </option>
          <option
            value="Benin"
          >
            Benin
             
            🇧🇯
          </option>
          <option
            value="Saint Barthelemy"
          >
            Saint Barthelemy
             
            🇧🇱
          </option>
          <option
            value="Bermuda"
          >
            Bermuda
             
            🇧🇲
          </option>
          <option
            value="Brunei Darussalam"
          >
            Brunei Darussalam
             
            🇧🇳
          </option>
          <option
            value="Bolivia"
          >
            Bolivia
             
            🇧🇴
          </option>
          <option
            value="Brazil"
          >
            Brazil
             
            🇧🇷
          </option>
          <option
            value="Bahamas"
          >
            Bahamas
             
            🇧🇸
          </option>
          <option
            value="Bhutan"
          >
            Bhutan
             
            🇧🇹
          </option>
          <option
            value="Bouvet Island"
          >
            Bouvet Island
             
            🇧🇻
          </option>
          <option
            value="Botswana"
          >
            Botswana
             
            🇧🇼
          </option>
          <option
            value="Belarus"
          >
            Belarus
             
            🇧🇾
          </option>
          <option
            value="Belize"
          >
            Belize
             
            🇧🇿
          </option>
          <option
            value="Canada"
          >
            Canada
             
            🇨🇦
          </option>
          <option
            value="Cocos (Keeling) Islands"
          >
            Cocos (Keeling) Islands
             
            🇨🇨
          </option>
          <option
            value="Congo, Republic of the"
          >
            Congo, Republic of the
             
            🇨🇩
          </option>
          <option
            value="Central African Republic"
          >
            Central African Republic
             
            🇨🇫
          </option>
          <option
            value="Congo, Democratic Republic of the"
          >
            Congo, Democratic Republic of the
             
            🇨🇬
          </option>
          <option
            value="Switzerland"
          >
            Switzerland
             
            🇨🇭
          </option>
          <option
            value="Cote d'Ivoire"
          >
            Cote d'Ivoire
             
            🇨🇮
          </option>
          <option
            value="Cook Islands"
          >
            Cook Islands
             
            🇨🇰
          </option>
          <option
            value="Chile"
          >
            Chile
             
            🇨🇱
          </option>
          <option
            value="Cameroon"
          >
            Cameroon
             
            🇨🇲
          </option>
          <option
            value="China"
          >
            China
             
            🇨🇳
          </option>
          <option
            value="Colombia"
          >
            Colombia
             
            🇨🇴
          </option>
          <option
            value="Costa Rica"
          >
            Costa Rica
             
            🇨🇷
          </option>
          <option
            value="Cuba"
          >
            Cuba
             
            🇨🇺
          </option>
          <option
            value="Cape Verde"
          >
            Cape Verde
             
            🇨🇻
          </option>
          <option
            value="Curacao"
          >
            Curacao
             
            🇨🇼
          </option>
          <option
            value="Christmas Island"
          >
            Christmas Island
             
            🇨🇽
          </option>
          <option
            value="Cyprus"
          >
            Cyprus
             
            🇨🇾
          </option>
          <option
            value="Czech Republic"
          >
            Czech Republic
             
            🇨🇿
          </option>
          <option
            value="Germany"
          >
            Germany
             
            🇩🇪
          </option>
          <option
            value="Djibouti"
          >
            Djibouti
             
            🇩🇯
          </option>
          <option
            value="Denmark"
          >
            Denmark
             
            🇩🇰
          </option>
          <option
            value="Dominica"
          >
            Dominica
             
            🇩🇲
          </option>
          <option
            value="Dominican Republic"
          >
            Dominican Republic
             
            🇩🇴
          </option>
          <option
            value="Algeria"
          >
            Algeria
             
            🇩🇿
          </option>
          <option
            value="Ecuador"
          >
            Ecuador
             
            🇪🇨
          </option>
          <option
            value="Estonia"
          >
            Estonia
             
            🇪🇪
          </option>
          <option
            value="Egypt"
          >
            Egypt
             
            🇪🇬
          </option>
          <option
            value="Western Sahara"
          >
            Western Sahara
             
            🇪🇭
          </option>
          <option
            value="Eritrea"
          >
            Eritrea
             
            🇪🇷
          </option>
          <option
            value="Spain"
          >
            Spain
             
            🇪🇸
          </option>
          <option
            value="Ethiopia"
          >
            Ethiopia
             
            🇪🇹
          </option>
          <option
            value="Finland"
          >
            Finland
             
            🇫🇮
          </option>
          <option
            value="Fiji"
          >
            Fiji
             
            🇫🇯
          </option>
          <option
            value="Falkland Islands (Malvinas)"
          >
            Falkland Islands (Malvinas)
             
            🇫🇰
          </option>
          <option
            value="Micronesia, Federated States of"
          >
            Micronesia, Federated States of
             
            🇫🇲
          </option>
          <option
            value="Faroe Islands"
          >
            Faroe Islands
             
            🇫🇴
          </option>
          <option
            value="France"
          >
            France
             
            🇫🇷
          </option>
          <option
            value="Gabon"
          >
            Gabon
             
            🇬🇦
          </option>
          <option
            value="United Kingdom"
          >
            United Kingdom
             
            🇬🇧
          </option>
          <option
            value="Grenada"
          >
            Grenada
             
            🇬🇩
          </option>
          <option
            value="Georgia"
          >
            Georgia
             
            🇬🇪
          </option>
          <option
            value="French Guiana"
          >
            French Guiana
             
            🇬🇫
          </option>
          <option
            value="Guernsey"
          >
            Guernsey
             
            🇬🇬
          </option>
          <option
            value="Ghana"
          >
            Ghana
             
            🇬🇭
          </option>
          <option
            value="Gibraltar"
          >
            Gibraltar
             
            🇬🇮
          </option>
          <option
            value="Greenland"
          >
            Greenland
             
            🇬🇱
          </option>
          <option
            value="Gambia"
          >
            Gambia
             
            🇬🇲
          </option>
          <option
            value="Guinea"
          >
            Guinea
             
            🇬🇳
          </option>
          <option
            value="Guadeloupe"
          >
            Guadeloupe
             
            🇬🇵
          </option>
          <option
            value="Equatorial Guinea"
          >
            Equatorial Guinea
             
            🇬🇶
          </option>
          <option
            value="Greece"
          >
            Greece
             
            🇬🇷
          </option>
          <option
            value="South Georgia and the South Sandwich Islands"
          >
            South Georgia and the South Sandwich Islands
             
            🇬🇸
          </option>
          <option
            value="Guatemala"
          >
            Guatemala
             
            🇬🇹
          </option>
          <option
            value="Guam"
          >
            Guam
             
            🇬🇺
          </option>
          <option
            value="Guinea-Bissau"
          >
            Guinea-Bissau
             
            🇬🇼
          </option>
          <option
            value="Guyana"
          >
            Guyana
             
            🇬🇾
          </option>
          <option
            value="Hong Kong"
          >
            Hong Kong
             
            🇭🇰
          </option>
          <option
            value="Heard Island and McDonald Islands"
          >
            Heard Island and McDonald Islands
             
            🇭🇲
          </option>
          <option
            value="Honduras"
          >
            Honduras
             
            🇭🇳
          </option>
          <option
            value="Croatia"
          >
            Croatia
             
            🇭🇷
          </option>
          <option
            value="Haiti"
          >
            Haiti
             
            🇭🇹
          </option>
          <option
            value="Hungary"
          >
            Hungary
             
            🇭🇺
          </option>
          <option
            value="Indonesia"
          >
            Indonesia
             
            🇮🇩
          </option>
          <option
            value="Ireland"
          >
            Ireland
             
            🇮🇪
          </option>
          <option
            value="Israel"
          >
            Israel
             
            🇮🇱
          </option>
          <option
            value="Isle of Man"
          >
            Isle of Man
             
            🇮🇲
          </option>
          <option
            value="India"
          >
            India
             
            🇮🇳
          </option>
          <option
            value="British Indian Ocean Territory"
          >
            British Indian Ocean Territory
             
            🇮🇴
          </option>
          <option
            value="Iraq"
          >
            Iraq
             
            🇮🇶
          </option>
          <option
            value="Iran, Islamic Republic of"
          >
            Iran, Islamic Republic of
             
            🇮🇷
          </option>
          <option
            value="Iceland"
          >
            Iceland
             
            🇮🇸
          </option>
          <option
            value="Italy"
          >
            Italy
             
            🇮🇹
          </option>
          <option
            value="Jersey"
          >
            Jersey
             
            🇯🇪
          </option>
          <option
            value="Jamaica"
          >
            Jamaica
             
            🇯🇲
          </option>
          <option
            value="Jordan"
          >
            Jordan
             
            🇯🇴
          </option>
          <option
            value="Japan"
          >
            Japan
             
            🇯🇵
          </option>
          <option
            value="Kenya"
          >
            Kenya
             
            🇰🇪
          </option>
          <option
            value="Kyrgyzstan"
          >
            Kyrgyzstan
             
            🇰🇬
          </option>
          <option
            value="Cambodia"
          >
            Cambodia
             
            🇰🇭
          </option>
          <option
            value="Kiribati"
          >
            Kiribati
             
            🇰🇮
          </option>
          <option
            value="Comoros"
          >
            Comoros
             
            🇰🇲
          </option>
          <option
            value="Saint Kitts and Nevis"
          >
            Saint Kitts and Nevis
             
            🇰🇳
          </option>
          <option
            value="Korea, Democratic People's Republic of"
          >
            Korea, Democratic People's Republic of
             
            🇰🇵
          </option>
          <option
            value="Korea, Republic of"
          >
            Korea, Republic of
             
            🇰🇷
          </option>
          <option
            value="Kuwait"
          >
            Kuwait
             
            🇰🇼
          </option>
          <option
            value="Cayman Islands"
          >
            Cayman Islands
             
            🇰🇾
          </option>
          <option
            value="Kazakhstan"
          >
            Kazakhstan
             
            🇰🇿
          </option>
          <option
            value="Lao People's Democratic Republic"
          >
            Lao People's Democratic Republic
             
            🇱🇦
          </option>
          <option
            value="Lebanon"
          >
            Lebanon
             
            🇱🇧
          </option>
          <option
            value="Saint Lucia"
          >
            Saint Lucia
             
            🇱🇨
          </option>
          <option
            value="Liechtenstein"
          >
            Liechtenstein
             
            🇱🇮
          </option>
          <option
            value="Sri Lanka"
          >
            Sri Lanka
             
            🇱🇰
          </option>
          <option
            value="Liberia"
          >
            Liberia
             
            🇱🇷
          </option>
          <option
            value="Lesotho"
          >
            Lesotho
             
            🇱🇸
          </option>
          <option
            value="Lithuania"
          >
            Lithuania
             
            🇱🇹
          </option>
          <option
            value="Luxembourg"
          >
            Luxembourg
             
            🇱🇺
          </option>
          <option
            value="Latvia"
          >
            Latvia
             
            🇱🇻
          </option>
          <option
            value="Libya"
          >
            Libya
             
            🇱🇾
          </option>
          <option
            value="Morocco"
          >
            Morocco
             
            🇲🇦
          </option>
          <option
            value="Monaco"
          >
            Monaco
             
            🇲🇨
          </option>
          <option
            value="Moldova, Republic of"
          >
            Moldova, Republic of
             
            🇲🇩
          </option>
          <option
            value="Montenegro"
          >
            Montenegro
             
            🇲🇪
          </option>
          <option
            value="Saint Martin (French part)"
          >
            Saint Martin (French part)
             
            🇲🇫
          </option>
          <option
            value="Madagascar"
          >
            Madagascar
             
            🇲🇬
          </option>
          <option
            value="Marshall Islands"
          >
            Marshall Islands
             
            🇲🇭
          </option>
          <option
            value="Macedonia, the Former Yugoslav Republic of"
          >
            Macedonia, the Former Yugoslav Republic of
             
            🇲🇰
          </option>
          <option
            value="Mali"
          >
            Mali
             
            🇲🇱
          </option>
          <option
            value="Myanmar"
          >
            Myanmar
             
            🇲🇲
          </option>
          <option
            value="Mongolia"
          >
            Mongolia
             
            🇲🇳
          </option>
          <option
            value="Macao"
          >
            Macao
             
            🇲🇴
          </option>
          <option
            value="Northern Mariana Islands"
          >
            Northern Mariana Islands
             
            🇲🇵
          </option>
          <option
            value="Martinique"
          >
            Martinique
             
            🇲🇶
          </option>
          <option
            value="Mauritania"
          >
            Mauritania
             
            🇲🇷
          </option>
          <option
            value="Montserrat"
          >
            Montserrat
             
            🇲🇸
          </option>
          <option
            value="Malta"
          >
            Malta
             
            🇲🇹
          </option>
          <option
            value="Mauritius"
          >
            Mauritius
             
            🇲🇺
          </option>
          <option
            value="Maldives"
          >
            Maldives
             
            🇲🇻
          </option>
          <option
            value="Malawi"
          >
            Malawi
             
            🇲🇼
          </option>
          <option
            value="Mexico"
          >
            Mexico
             
            🇲🇽
          </option>
          <option
            value="Malaysia"
          >
            Malaysia
             
            🇲🇾
          </option>
          <option
            value="Mozambique"
          >
            Mozambique
             
            🇲🇿
          </option>
          <option
            value="Namibia"
          >
            Namibia
             
            🇳🇦
          </option>
          <option
            value="New Caledonia"
          >
            New Caledonia
             
            🇳🇨
          </option>
          <option
            value="Niger"
          >
            Niger
             
            🇳🇪
          </option>
          <option
            value="Norfolk Island"
          >
            Norfolk Island
             
            🇳🇫
          </option>
          <option
            value="Nigeria"
          >
            Nigeria
             
            🇳🇬
          </option>
          <option
            value="Nicaragua"
          >
            Nicaragua
             
            🇳🇮
          </option>
          <option
            value="Netherlands"
          >
            Netherlands
             
            🇳🇱
          </option>
          <option
            value="Norway"
          >
            Norway
             
            🇳🇴
          </option>
          <option
            value="Nepal"
          >
            Nepal
             
            🇳🇵
          </option>
          <option
            value="Nauru"
          >
            Nauru
             
            🇳🇷
          </option>
          <option
            value="Niue"
          >
            Niue
             
            🇳🇺
          </option>
          <option
            value="New Zealand"
          >
            New Zealand
             
            🇳🇿
          </option>
          <option
            value="Oman"
          >
            Oman
             
            🇴🇲
          </option>
          <option
            value="Panama"
          >
            Panama
             
            🇵🇦
          </option>
          <option
            value="Peru"
          >
            Peru
             
            🇵🇪
          </option>
          <option
            value="French Polynesia"
          >
            French Polynesia
             
            🇵🇫
          </option>
          <option
            value="Papua New Guinea"
          >
            Papua New Guinea
             
            🇵🇬
          </option>
          <option
            value="Philippines"
          >
            Philippines
             
            🇵🇭
          </option>
          <option
            value="Pakistan"
          >
            Pakistan
             
            🇵🇰
          </option>
          <option
            value="Poland"
          >
            Poland
             
            🇵🇱
          </option>
          <option
            value="Saint Pierre and Miquelon"
          >
            Saint Pierre and Miquelon
             
            🇵🇲
          </option>
          <option
            value="Pitcairn"
          >
            Pitcairn
             
            🇵🇳
          </option>
          <option
            value="Puerto Rico"
          >
            Puerto Rico
             
            🇵🇷
          </option>
          <option
            value="Palestine, State of"
          >
            Palestine, State of
             
            🇵🇸
          </option>
          <option
            value="Portugal"
          >
            Portugal
             
            🇵🇹
          </option>
          <option
            value="Palau"
          >
            Palau
             
            🇵🇼
          </option>
          <option
            value="Paraguay"
          >
            Paraguay
             
            🇵🇾
          </option>
          <option
            value="Qatar"
          >
            Qatar
             
            🇶🇦
          </option>
          <option
            value="Reunion"
          >
            Reunion
             
            🇷🇪
          </option>
          <option
            value="Romania"
          >
            Romania
             
            🇷🇴
          </option>
          <option
            value="Serbia"
          >
            Serbia
             
            🇷🇸
          </option>
          <option
            value="Russian Federation"
          >
            Russian Federation
             
            🇷🇺
          </option>
          <option
            value="Rwanda"
          >
            Rwanda
             
            🇷🇼
          </option>
          <option
            value="Saudi Arabia"
          >
            Saudi Arabia
             
            🇸🇦
          </option>
          <option
            value="Solomon Islands"
          >
            Solomon Islands
             
            🇸🇧
          </option>
          <option
            value="Seychelles"
          >
            Seychelles
             
            🇸🇨
          </option>
          <option
            value="Sudan"
          >
            Sudan
             
            🇸🇩
          </option>
          <option
            value="Sweden"
          >
            Sweden
             
            🇸🇪
          </option>
          <option
            value="Singapore"
          >
            Singapore
             
            🇸🇬
          </option>
          <option
            value="Saint Helena"
          >
            Saint Helena
             
            🇸🇭
          </option>
          <option
            value="Slovenia"
          >
            Slovenia
             
            🇸🇮
          </option>
          <option
            value="Svalbard and Jan Mayen"
          >
            Svalbard and Jan Mayen
             
            🇸🇯
          </option>
          <option
            value="Slovakia"
          >
            Slovakia
             
            🇸🇰
          </option>
          <option
            value="Sierra Leone"
          >
            Sierra Leone
             
            🇸🇱
          </option>
          <option
            value="San Marino"
          >
            San Marino
             
            🇸🇲
          </option>
          <option
            value="Senegal"
          >
            Senegal
             
            🇸🇳
          </option>
          <option
            value="Somalia"
          >
            Somalia
             
            🇸🇴
          </option>
          <option
            value="Suriname"
          >
            Suriname
             
            🇸🇷
          </option>
          <option
            value="South Sudan"
          >
            South Sudan
             
            🇸🇸
          </option>
          <option
            value="Sao Tome and Principe"
          >
            Sao Tome and Principe
             
            🇸🇹
          </option>
          <option
            value="El Salvador"
          >
            El Salvador
             
            🇸🇻
          </option>
          <option
            value="Sint Maarten (Dutch part)"
          >
            Sint Maarten (Dutch part)
             
            🇸🇽
          </option>
          <option
            value="Syrian Arab Republic"
          >
            Syrian Arab Republic
             
            🇸🇾
          </option>
          <option
            value="Swaziland"
          >
            Swaziland
             
            🇸🇿
          </option>
          <option
            value="Turks and Caicos Islands"
          >
            Turks and Caicos Islands
             
            🇹🇨
          </option>
          <option
            value="Chad"
          >
            Chad
             
            🇹🇩
          </option>
          <option
            value="French Southern Territories"
          >
            French Southern Territories
             
            🇹🇫
          </option>
          <option
            value="Togo"
          >
            Togo
             
            🇹🇬
          </option>
          <option
            value="Thailand"
          >
            Thailand
             
            🇹🇭
          </option>
          <option
            value="Tajikistan"
          >
            Tajikistan
             
            🇹🇯
          </option>
          <option
            value="Tokelau"
          >
            Tokelau
             
            🇹🇰
          </option>
          <option
            value="Timor-Leste"
          >
            Timor-Leste
             
            🇹🇱
          </option>
          <option
            value="Turkmenistan"
          >
            Turkmenistan
             
            🇹🇲
          </option>
          <option
            value="Tunisia"
          >
            Tunisia
             
            🇹🇳
          </option>
          <option
            value="Tonga"
          >
            Tonga
             
            🇹🇴
          </option>
          <option
            value="Turkey"
          >
            Turkey
             
            🇹🇷
          </option>
          <option
            value="Trinidad and Tobago"
          >
            Trinidad and Tobago
             
            🇹🇹
          </option>
          <option
            value="Tuvalu"
          >
            Tuvalu
             
            🇹🇻
          </option>
          <option
            value="Taiwan, Province of China"
          >
            Taiwan, Province of China
             
            🇹🇼
          </option>
          <option
            value="United Republic of Tanzania"
          >
            United Republic of Tanzania
             
            🇹🇿
          </option>
          <option
            value="Ukraine"
          >
            Ukraine
             
            🇺🇦
          </option>
          <option
            value="Uganda"
          >
            Uganda
             
            🇺🇬
          </option>
          <option
            value="United States"
          >
            United States
             
            🇺🇸
          </option>
          <option
            value="Uruguay"
          >
            Uruguay
             
            🇺🇾
          </option>
          <option
            value="Uzbekistan"
          >
            Uzbekistan
             
            🇺🇿
          </option>
          <option
            value="Holy See (Vatican City State)"
          >
            Holy See (Vatican City State)
             
            🇻🇦
          </option>
          <option
            value="Saint Vincent and the Grenadines"
          >
            Saint Vincent and the Grenadines
             
            🇻🇨
          </option>
          <option
            value="Venezuela"
          >
            Venezuela
             
            🇻🇪
          </option>
          <option
            value="British Virgin Islands"
          >
            British Virgin Islands
             
            🇻🇬
          </option>
          <option
            value="US Virgin Islands"
          >
            US Virgin Islands
             
            🇻🇮
          </option>
          <option
            value="Vietnam"
          >
            Vietnam
             
            🇻🇳
          </option>
          <option
            value="Vanuatu"
          >
            Vanuatu
             
            🇻🇺
          </option>
          <option
            value="Wallis and Futuna"
          >
            Wallis and Futuna
             
            🇼🇫
          </option>
          <option
            value="Samoa"
          >
            Samoa
             
            🇼🇸
          </option>
          <option
            value="Kosovo"
          >
            Kosovo
             
            🇽🇰
          </option>
          <option
            value="Yemen"
          >
            Yemen
             
            🇾🇪
          </option>
          <option
            value="Mayotte"
          >
            Mayotte
             
            🇾🇹
          </option>
          <option
            value="South Africa"
          >
            South Africa
             
            🇿🇦
          </option>
          <option
            value="Zambia"
          >
            Zambia
             
            🇿🇲
          </option>
          <option
            value="Zimbabwe"
          >
            Zimbabwe
             
            🇿🇼
          </option>
        </select>
      </section>
      <section
        className="date-container"
      >
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="date-picker"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Select date"
              readOnly={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </section>
      <section
        className="options-container"
      >
        <input
          name="number_of_days"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="🏖 Number of days"
        />
        <input
          name="number_of_rooms"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="🏘 Number of rooms"
        />
        <input
          name="number_of_guests"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="👥 Number of guests"
        />
      </section>
      <section
        className="select-hotel-container"
      >
        <label>
          Select Hotel:
        </label>
        <select
          name="hotel_id"
          onBlur={[Function]}
          onChange={[Function]}
        />
      </section>
      <section
        className="submit-container"
      >
        <input
          type="submit"
          value="RESERVE"
        />
      </section>
    </form>
  </div>
</div>
`;
